dm/dm/config/subtask.go

```diff
@@ -394,10 +395,23 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		c.MetaSchema = defaultMetaSchema
 	}
 
-	dirSuffix := "." + c.Name
-	if !strings.HasSuffix(c.LoaderConfig.Dir, dirSuffix) { // check to support multiple times calling
-		// if not ends with the task name, we append the task name to the tail
-		c.LoaderConfig.Dir += dirSuffix
+	// adjust dir
+	if c.Mode == ModeAll || c.Mode == ModeFull {
+		isS3, newDir, err := exstorage.AdjustS3Path(c.LoaderConfig.Dir, c.Name+"."+c.SourceID)
+		if err != nil {
```

Comment:
this may mean it's a local directory

Reply:
yes, if it is  a local directory it will not be  adjusted. i will try to change logic like comment at line 414.

Reply:
When user set `./dumpdir`, `IsS3Path` will return error? Also those function should add unit tests for local paths like `./dumpdir`.

I think it's still not fixed, please don't mark resolved on your own.

Reply:
ok, sorry, i will not resolve by myself. because `utils` add smore functions, i will add more ut to `dm/pkg/exstorage/utils.go` and just save the conversation about add tests. by the way, dir like `./dumpdir` will not return error.